resources/internal.rb

```diff
@@ -1,76 +1,76 @@
 # need build_resource
-include Chef::DSL::Recipe
 
 default_action :install
 
 provides :multipackage_internal
 
-property :package_name, [String, Array], :name_property => true
+property :package_name, [String, Array], name_property: true
 property :version, [String, Array]
 property :options, String
 property :timeout, [String, Integer]
 
 action :install do
-  do_action(new_resource, :install)
+  do_action(:install)
 end
 
 action :upgrade do
-  do_action(new_resource, :upgrade)
+  do_action(:upgrade)
 end
 
 action :reconfig do
-  do_action(new_resource, :reconfig)
+  do_action(:reconfig)
 end
 
 action :remove do
-  do_action(new_resource, :remove)
+  do_action(:remove)
 end
 
 action :purge do
-  do_action(new_resource, :purge)
+  do_action(:purge)
 end
 
-# FIXME: can we get rid of new_resource everywhere?
-def multipackage_resource(new_resource, action)
-  package_name_array = package_name.is_a?(Array) ? package_name : package_name.split(", ")
-  version_array = [new_resource.version].flatten if new_resource.version
-  package new_resource.name do
-    package_name package_name_array
-    version version_array if version_array
-    options new_resource.options if new_resource.options
-    timeout new_resource.timeout if new_resource.timeout
-    action action
-  end
-end
-
-def singlepackage_resources(new_resource, action)
-  package_name_array = package_name.is_a?(Array) ? package_name : package_name.split(", ")
-  version_array = [new_resource.version].flatten if new_resource.version
-  package_name_array.each_with_index do |package_name, i|
-    version = version_array[i] if version_array
-    package package_name do
-      version version if version
+action_class.class_eval <<-EOM, __FILE__, __LINE__+1
```

Comment:
keiser likes this way better than the define_method one because you get better line numbers in the stack trace.

really this needs to be baked into the DSL so you can do some kind of `action_class_helper do ... end` block.